src/pages/en/guides/migrate-to-astro/from-gatsby.mdx

```diff
@@ -2,67 +2,884 @@
+- [MPA vs SPA](/en/concepts/mpa-vs-spa/): Gatsby is a React single-page app, and uses `index.js` as your project's root. Astro is a multi-page site, and `index.astro` is your home page.
 
 - [Astro components](/en/core-concepts/astro-components/) are not written as exported functions that return page templating. Instead, you'll split your code into a "code fence" and a body exclusively for the HTML you generate.
 
-- [Local file data](/en/guides/imports/): Gatsby uses GraphQL to retrieve data from your project files. Astro uses ESM imports and the [`Astro.glob()`](/en/guides/imports/#astroglob) helper to import data from your project files. GraphQL may be optionally added to your Astro project, but is not included by default.
+- [Local file data](/en/guides/imports/): Gatsby uses GraphQL to retrieve data from your project files. Astro uses ESM imports and a top-level await [`Astro.glob()`](/en/guides/imports/#astroglob) call to import data from your project files. GraphQL may be optionally be added to your Astro project, but is not included by default.
 
-## Switch from Gatsby to Astro
 
-To convert a Gatsby blog to Astro, start with our official blog theme starter template, or explore more community blog themes in our [theme showcase](https://astro.build/themes/). 
+## Convert your Project
 
-You can pass a `--template` argument to the `create-astro` command to start a new Astro project from any existing Astro repository on GitHub:
+### Get Started with Astro
 
-  <PackageManagerTabs>
-    <Fragment slot="npm">
-    ```shell
-    npm create astro -- --template withastro/blog
-    ```
-    </Fragment>
-    <Fragment slot="pnpm">
-    ```shell
-    pnpm create astro --template withastro/blog
-    ```
-    </Fragment>
-    <Fragment slot="yarn">
-    ```shell
-    yarn create astro --template withastro/blog
+- Option 1: **Create a new Astro project** 
+
+  Use `npm create astro@latest` to launch Astro's CLI wizard or choose a community theme from the [Astro Theme Showcase](https://astro.build/themes). 
+  
+  
+  You can also pass a `--template` argument to the `create-astro` command to start from an existing Astro repository on GitHub:
+
+    <PackageManagerTabs>
+      <Fragment slot="npm">
+      ```shell
+      npm create astro -- --template <github-username>/<github-repo>
+      ```
+      </Fragment>
+      <Fragment slot="pnpm">
+      ```shell
+      pnpm create astro --template <github-username>/<github-repo>
+      ```
+      </Fragment>
+      <Fragment slot="yarn">
+      ```shell
+      yarn create astro --template <github-username>/<github-repo>
+      ```
+      </Fragment>
+    </PackageManagerTabs>
+  
+  Then, copy your existing Gatsby project files over to your new Astro project into a separate folder outside of `src`.
+
+- Option 2: **Update your existing Gatsby project dependencies**
```

Comment:
Does this really work? There are so many details that will not work that it is really hard to get stuff going again once you rip out the one and add the other. Also, "any Gatsby-related dependencies" is very vague especially with "Astro has support for installing NPM packages, including several for React. You may be able to keep some or all of your existing React components and dependencies." given.

I feel like you would want to promote only option 1 and make that a great experience instead of this part…

Gatsby is so entangled with the way GraphQL works that just removing it and adding Astro should nearly always result in a project that cannot start.

On the other hand, with option 1 you have something that can be viewed in the browser and than migrated file by file.